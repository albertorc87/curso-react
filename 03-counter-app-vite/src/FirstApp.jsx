import PropTypes from 'prop-types'

export const FirstApp = ({title, subTitle, name}) => {
    return (
        <>
            {/* <div data-testid="test-title">{title}</div> */}
            <h1>{title}</h1>
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{name}</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    subTitle: 'Subtitulo',
    name: 'Alber',
}