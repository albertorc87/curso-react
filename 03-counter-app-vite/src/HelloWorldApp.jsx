export default () => {
    return <h1>Hola chaico</h1>
}