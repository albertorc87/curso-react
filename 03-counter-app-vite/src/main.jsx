import React from 'react'
import ReactDom from 'react-dom/client'

import HelloWorldApp from './HelloWorldApp'
import {FirstApp} from './FirstApp'
import {CounterApp} from './CounterApp'

import './styles.css'


ReactDom.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={20}/>
        {/* <FirstApp title="Vamos a generar billetes ostias"/> */}
    </React.StrictMode>
)