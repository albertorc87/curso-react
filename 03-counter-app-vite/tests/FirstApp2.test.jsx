import {render, screen} from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'

describe('Pruebas en <FirstApp/>', () => {

    const title = "Vamos a generar billetes ostias"
    const subTitle = "Soy un subtitulo"

    test('debe hacer match con el snapshot', () => {
        const {container} = render(<FirstApp title={title}/>)
    
        expect(container).toMatchSnapshot()
    })

    test('debe mostrar el mensaje "Vamos a generar billetes ostias"', () => {
        render(<FirstApp title={title}/>)
        expect(screen.getByText(title)).toBeTruthy()
        // expect(screen.getByText(title)).not.toBeTruthy()
    })

    test('debe mostrar el titulo en un h1', () => {
        render(<FirstApp title={title}/>)
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title)
    })

    test('debe mostrar el subtitulo enviado por props', () => {
        render(<FirstApp 
            title={title}
            subTitle={subTitle}
        />)
    
        expect(screen.getAllByText(subTitle).length).toBe(2)
    })
})
