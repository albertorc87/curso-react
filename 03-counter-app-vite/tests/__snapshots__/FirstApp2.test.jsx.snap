// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Vamos a generar billetes ostias
  </h1>
  <p>
    Subtitulo
  </p>
  <p>
    Subtitulo
  </p>
  <p>
    Alber
  </p>
</div>
`;
